agent/consul/leader_connect_ca_test.go

```diff
@@ -778,37 +826,62 @@ func TestCAManager_Initialize_Vault_WithExternalTrustedCA(t *testing.T) {
 
```

Comment:
For the record `TestCAManager_Initialize_Vault_WithExternalTrustedCA` was a test I invested a lot in because it surfaced a bunch of strange things such as https://github.com/hashicorp/consul/pull/15661